lib/policyEvaluator/evaluator.js

```diff
@@ -0,0 +1,257 @@
+'use strict'; // eslint-disable-line strict
+
+const substituteVariables = require('./utils/variables.js');
+const handleWildcards = require('./utils/wildcards.js').handleWildcards;
+const conditions = require('./utils/conditions.js');
+const findConditionKey = conditions.findConditionKey;
+const convertConditionOperator = conditions.convertConditionOperator;
+const checkArnMatch = require('./utils/checkArnMatch.js');
+
+const evaluators = {};
+
+
+/**
+ * Check whether resource in policy statement applies to request resource
+ * @param {object} requestContext - info about request
+ * @param {string | [string]} statementResource - Resource(s) impacted
+ * by policy statement
+ * @return {boolean} true if applicable, false if not
+ */
+function isResourceApplicable(requestContext, statementResource) {
+    const resource = requestContext.resource;
+    if (!Array.isArray(statementResource)) {
+        // eslint-disable-next-line no-param-reassign
+        statementResource = [statementResource];
+    }
+    // ARN format:
+    // arn:partition:service:region:namespace:relative-id
+    const requestResourceArr = resource.split(':');
+    // Pull just the relative id because there is no restriction that it
+    // does not contain ":"
+    const requestRelativeId = requestResourceArr.slice(5).join(':');
+    for (let i = 0; i < statementResource.length; i ++) {
+        // Handle variables (must handle BEFORE wildcards)
+        const policyResource =
+            substituteVariables(statementResource[i], requestContext);
+        // Handle wildcards
+        const arnSegmentsMatch =
+            checkArnMatch(policyResource, requestRelativeId,
+                requestResourceArr, true);
+        if (arnSegmentsMatch) {
+            return true;
+        }
+        continue;
+    }
+    // If no match found, no resource is applicable
+    return false;
+}
+
+/**
+ * Check whether action in policy statement applies to request
+ * @param {Object} requestAction - Type of client request
+ * @param {string | [string]} statementAction - Action(s) impacted
+ * by policy statement
+ * @return {boolean} true if applicable, false if not
+ */
+function isActionApplicable(requestAction, statementAction) {
+    if (!Array.isArray(statementAction)) {
+        // eslint-disable-next-line no-param-reassign
+        statementAction = [statementAction];
+    }
+    for (let i = 0; i < statementAction.length; i ++) {
```

Comment:
For performance, i'm to keep that way, but store the length before to avoid object access :)